include/taskolib/Executor.h

```diff
@@ -198,6 +191,17 @@ class Executor
      */
     static VariableTable execute_sequence(Sequence sequence, Context context,
         std::shared_ptr<CommChannel> comm_channel) noexcept;
+
+    /**
+     * Determine if the executor is currently running a sequence in a separate thread.
+     *
+     * \returns true if a sequence is being executed or false otherwise.
+     *
+     * \note
+     * This function does not retrieve any messages from the message queue, but it does
+     * join the worker thread if it has finished its sequence.
```

Comment:
Is this the real problem? Are the messages irretrievable after `is_busy()` turned `false`?

Reply:
No, the messages would still be retrievable. However, it is way too tempting for users to *just* call `is_busy()` and forget about `update()`. In a way, this is exactly what happened in the Taskomat server, where we called `is_busy()` in the RUNNING property, but left `update()` to the periodical `EqFct::update()` call. Consequence: Once RUNNING becomes 0, an external observer can look at the state of the server and still find the non-updated state of the sequence.